hazelcast/src/main/java/com/hazelcast/collection/impl/txncollection/CollectionTransactionLogRecord.java

```diff
@@ -58,52 +50,26 @@ public CollectionTransactionLogRecord(long itemId,
 
     @Override
     public Object getKey() {
-        return new TransactionRecordKey(name, itemId, serviceName);
+        return new TransactionLogRecordKey(name, itemId, serviceName);
     }
 
     @Override
-    public Future prepare(NodeEngine nodeEngine) {
+    public Operation createPrepareOperation() {
         boolean removeOperation = op instanceof CollectionTxnRemoveOperation;
-        CollectionPrepareOperation operation = new CollectionPrepareOperation(name, itemId, transactionId, removeOperation);
-        try {
-            return nodeEngine.getOperationService().invokeOnPartition(serviceName, operation, partitionId);
-        } catch (Throwable t) {
-            throw ExceptionUtil.rethrow(t);
-        }
-    }
-
-    @Override
-    public Future commit(NodeEngine nodeEngine) {
-        try {
-            return nodeEngine.getOperationService().invokeOnPartition(serviceName, op, partitionId);
-        } catch (Throwable t) {
-            throw ExceptionUtil.rethrow(t);
-        }
+        return new CollectionPrepareOperation(partitionId, name, serviceName, itemId, transactionId, removeOperation);
     }
 
     @Override
-    public void commitAsync(NodeEngine nodeEngine, ExecutionCallback callback) {
-        InternalOperationService operationService = (InternalOperationService) nodeEngine.getOperationService();
-        operationService.asyncInvokeOnPartition(serviceName, op, partitionId, callback);
-    }
-
-    @Override
-    public Future rollback(NodeEngine nodeEngine) {
-        boolean removeOperation = op instanceof CollectionTxnRemoveOperation;
-        CollectionRollbackOperation operation = new CollectionRollbackOperation(name, itemId, removeOperation);
-        try {
-            return nodeEngine.getOperationService().invokeOnPartition(serviceName, operation, partitionId);
-        } catch (Throwable t) {
-            throw ExceptionUtil.rethrow(t);
-        }
+    public Operation createCommitOperation() {
```

Comment:
I'll go for correct names. You apply consistency for good things, so that more good things are caused bu other good things. But you don't apply consistency for inappropriate naming choices.